### PR TITLE
Milestone picker: support up/down arrows #1412

### LINKED ISSUE #1412: Milestone picker: support up/down arrows
Let's say we want to reschedule an issue from current milestone to the next one. `M` `down` `enter` should accomplish that.

### PR DESCRIPTION
**Pull request by m133225**
Fixes #1412 .

### CODE REVIEW

File: src/main/java/ui/components/pickers/MilestonePickerDialog.java [hunk 1]
```diff
@@ -77,6 +78,19 @@ private void setupKeyEvents() {
...
             handleUpdatedInput(newValue);
         });
+        this.
+        inputField.setOnKeyPressed(e -> {
+            if (e.getCode() == KeyCode.DOWN) {
+                e.consume();
+                state.selectNextBestMatchingMilestone();
+                refreshUI(state);
+            }
+            if (e.getCode() == KeyCode.UP) {
```
Comment (damithc) [inline src/main/java/ui/components/pickers/MilestonePickerDialog.java]: `else if`? or `switch`?

**Comment by m133225** [comment]
Ready for review.

**Comment by ndt93** [comment]
@dariusf Hi Darius, can help review this?

File: src/main/java/ui/components/pickers/MilestonePickerState.java [hunk 2]
```diff
@@ -54,6 +54,50 @@ private void processMatchingMilestonesByQueryWord(List<PickerMilestone> mileston
...
+     * @param milestones
+     * @return -1 if there are no selected milestones
+     */
+    private int getSelectedIndex(List<PickerMilestone> milestones) {
+        for (int i = 0; i < milestones.size(); i++) {
+            if (milestones.get(i).isSelected()) {
+                return i;
+            }
+        }
+        return -1;
```
Comment (dariusf) [inline src/main/java/ui/components/pickers/MilestonePickerState.java]: Is it possible that nothing is selected? If so, the return type should be optional (-1 is kind of ad hoc). If it's not possible, there should be an assertion failure.

**Comment by m133225** [comment]
@dariusf Changed the method to return an optional!

**Pull request merged**